### PR TITLE
Fix async demo and adding some warnings on dependents

### PR DESCRIPTION
**Pull request by edgesoft**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+**/.DS_Store
```
Comment (Guria) [inline .gitignore]: this should live in your global gitignore https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

File: actions/validate.js [hunk 2]
```diff
@@ -22,12 +22,14 @@ function validate(arg) {
...
 
   if (Array.isArray(field.dependents) && field.dependents.length) {
+  	
     field.dependents.forEach(function(dependent) {
       var path = joinPath(input.field, dependent);
-      var key = path.pop();
-      var form = state.get(path);
-
-      doValidation(path, form, key);
+      if( path ){
```
Comment (Guria) [inline actions/validate.js]: hmm, is path here can be really falsy?
What happens if it falsy here? Just ignore dependent?

Comment (edgesoft) [inline actions/validate.js]: Yes, or maybe just do it and get a giant error. It's really easy to miss [[]] in the code and just do []

File: helpers/joinPath.js [hunk 3]
```diff
@@ -1,4 +1,10 @@
 module.exports = function joinPath (/*base, path1, path2, ... */) {
+
+  if( ! arguments[1] || ! 	Array.isArray(arguments[1]) ){
+  	console.warn( "arguments[1] dependent must be an array" );
```
Comment (Guria) [inline helpers/joinPath.js]: add a prefix to warning, like `cerebral-module-forms:` and make it less cryptic.
can you provide `joinPath` usage example to get an idea of how it works?

File: actions/validate.js (hunk 2 above)
Comment (Guria) [inline actions/validate.js]: I suppose if it falsy here it means some potential error here. Need some context on what we get here and when. Haven't used forms yet :(

Comment (edgesoft) [inline actions/validate.js]: Yes, @abalmos added this to be able to depend on other forms like formA: {} , formB: {} and be able to validate against other forms or field in the same form. It also updates the field if another one is validated. But there was a very cryptic error when I was using [] and forgot [[]] and it took me a while to see the bug. Therefor I added this so the next programmer didn't have to spend some time on this. But maybe it's better to just let it fail.

Comment (Guria) [inline actions/validate.js]: @edgesoft please provide example of part with `[[]]`

File: .gitignore (hunk 1 above)
Comment (edgesoft) [inline .gitignore]: This is now removed.

**Comment by Guria** [comment]
@edgesoft okay looks its better start from scratch here :)

**Comment by edgesoft** [comment]
Should I close the pull request and create another one?

**Pull request closed without merging**